skrf/vi/vna/hp/hp8510c.py

```diff
@@ -1,187 +1,419 @@
-        raw = self.query_values(cmd, container=np.array, delay=self.query_delay)[1:]
-        vals = raw.reshape((-1, 2))
-        vals_complex = (vals[:,0] + 1j * vals[:,1]).flatten()
-        return vals_complex
-
-    def _get_oneport(self, parameter: tuple[int, int], sweep: bool=True):
-        if any(p not in {1,2} for p in parameter):
-            raise ValueError("The elements of parameter must be 1, or 2.")
-
-        self.write(f"s{parameter[0]}{parameter[1]}")
-
-        if sweep:
-            self.sweep()
-
-        ntwk = skrf.Network(name=f"S{parameter[0]}{parameter[1]}")
-        ntwk.s = self.get_complex_data("OUTPDATA")
-        ntwk.frequency = self.frequency
-
-        return ntwk
-
-    def _get_twoport(self, sweep: bool=True) -> skrf.Network:
-        s11 = self._get_oneport((1, 1))
-        freq = s11.frequency
-        s11 = s11.s[:, 0, 0]
-        s12 = self._get_oneport((1, 2)).s[:, 0, 0]
-        s21 = self._get_oneport((2, 1)).s[:, 0, 0]
-        s22 = self._get_oneport((2, 2)).s[:, 0, 0]
-
-        ntwk = skrf.Network()
-        ntwk.s = np.array([
-            [s11, s21],
-            [s12, s22]
-        ]).transpose().reshape((-1, 2, 2))
-        ntwk.frequency = freq
-
-        return ntwk
-
-    def get_snp_network(self, ports: Sequence = {1,2}) -> skrf.Network:
-        ports = set(ports)
-        if not ports.issubset({1,2}):
-            raise ValueError("This instrument only has two ports. Must pass 1, 2, or (1,2)")
-
-        if len(ports) == 1:
-            p = ports[0]
-            return self._get_oneport((p, p))
-        else:
-            return self._get_twoport()
-            return self._get_twoport()
+"""
+.. module:: skrf.vi.vna.hp.8510c
+=================================================
+HP 8510C (:mod:`skrf.vi.vna.hp.8510c`)
+=================================================
+
+Instrument Class
+================
+
+.. autosummary::
+    :nosignatures:
+    :toctree: generated/
+
+Basic one-port:
+>>> vna = skrf.vi.vna.hp.HP8510C('TCPIP::ad007-right.lan::gpib0,16::INSTR')
+>>> vna.set_frequency_sweep(2e9,3e9,201)
+>>> vna.get_snp_network(ports=(1,))
+
+Basic two-port:
+>>> vna = skrf.vi.vna.hp.HP8510C('TCPIP::ad007-right.lan::gpib0,16::INSTR')
+>>> vna.set_frequency_sweep(2e9,3e9,201)
+>>> vna.get_snp_network(ports=(1,2))
+
+Intermediate example -- note that 1001 point sweeps are not natively supported by the instrument; this driver
+takes multiple sweeps and pastes the results together.
+>>> vna = skrf.vi.vna.hp.HP8510C('TCPIP::ad007-right.lan::gpib0,16::INSTR')
+>>> vna.set_frequency_sweep(2e9,3e9,1001)
+>>> vna.get_snp_network(ports=(1,2))
+
+Advanced example.
+>>> vna = skrf.vi.vna.hp.HP8510C('TCPIP::ad007-right.lan::gpib0,16::INSTR')
+>>> freq_block_1 = np.linspace(1e9,2e9,801)
+>>> freq_block_2 = [10e9,11e9,12e9]
+>>> freqs = np.concatenate((freq_block_1, freq_block_2))
+>>> vna.frequency = skrf.Frequency.from_f(freqs)
+>>> vna.get_snp_network(ports=(1,2))
+"""
+
+import numpy as npy

```

Comment:
@jhillairet I just noticed, it seems some parts of skrf use `npy` and some parts use `np`. Worth it to standardize on one or the other?

Reply:
The `npy` notation is historic. I guess we should now favor `np`. What do you think @FranzForstmayr ? We could homogeneize.

Reply:
I think we should go for `np`, but this could be a batch PR for all files.

Reply:
I personally prefer `np` and was only using `npy` to rhyme with other code I saw, so I would be happy to change the PR -- and consider this a vote for global `np` too :)

Reply:
PR updated (but just for my code, I didn't rope in the scope of a global search and replace)